sound/soc/sof/pm.c

```diff
@@ -119,8 +119,15 @@ static int sof_resume(struct device *dev, bool runtime_resume)
 	 * D0 substate.
 	 */
 	if (!runtime_resume && sof_ops(sdev)->set_power_state &&
-	    old_state == SOF_DSP_PM_D0)
+	    old_state == SOF_DSP_PM_D0) {
+		ret = snd_sof_dma_trace_trigger(sdev, SNDRV_PCM_TRIGGER_START);
```

Comment:
why not check if sdev->dtrace_is_enabled is already true before triggering again? It could have failed during suspend right? And then this would mean you don't need to do the check for it again in the platform-specific trigger as above in hda-trace.c